File: app/src/main/java/com/example/sarada/moviereviews/data/FavoriteContract.java
package com.example.sarada.moviereviews.data;

import android.net.Uri;
import android.provider.BaseColumns;

public class FavoriteContract {

    public static final String AUTHORITY = "com.example.sarada.moviereviews";
    public static final Uri BASE_CONTENT_URI = Uri.parse("content://" + AUTHORITY);
    public static final String PATH_FAVORITE = "favorite";

    public static final class FavoriteEntry implements BaseColumns {

        public static final Uri CONTENT_URI = BASE_CONTENT_URI.buildUpon().appendPath(PATH_FAVORITE).build();

        public static final String TABLE_NAME = "favorite";
        public static final String COLUMN_MOVIEID = "movieid";
        public static final String COLUMN_TITLE = "title";
        public static final String COLUMN_USERRATING = "userrating";
        public static final String COLUMN_POSTER_PATH = "posterpath";
        public static final String COLUMN_PLOT_SYNOPSIS = "overview";
        public static final String COLUMN_RELEASE_DATE = "date";
    }
}

File: app/src/main/java/com/example/sarada/moviereviews/database/FavoriteContract.kt
package com.example.sarada.moviereviews.database

import android.net.Uri
import android.provider.BaseColumns

object FavoriteContract {
    const val AUTHORITY = "com.example.sarada.moviereviews"
    val BASE_CONTENT_URI = Uri.parse("content://" + AUTHORITY)
    const val PATH_FAVORITE = "favorite"

    object FavoriteEntry : BaseColumns {
        @JvmField
        val CONTENT_URI = BASE_CONTENT_URI.buildUpon().appendPath(PATH_FAVORITE).build()
        const val _ID = "_id"
        const val TABLE_NAME = "favorite"
        const val COLUMN_MOVIEID = "movieid"
        const val COLUMN_TITLE = "title"
        const val COLUMN_USERRATING = "userrating"
        const val COLUMN_POSTER_PATH = "posterpath"
        const val COLUMN_PLOT_SYNOPSIS = "overview"
        const val COLUMN_RELEASE_DATE = "date"
    }
}